ios updates (#1389)
Appboy/braze-docs: _docs/_developer_guide/platform_integration_guides/ios/ios_14.md

# iOS 14 SDK Upgrade Guide

This guide describes Braze-related changes introduced in iOS 14 and the required upgrade steps for your Braze iOS SDK integration.

For a complete list of new iOS 14 updates announced this year at WWDC, see Apple's [iOS 14 Preview](https://www.apple.com/ios/ios-14-preview/).

#### Summary of iOS 14 breaking changes

- Geofences are no longer supported by iOS for users who choose the new  _approximate location_ permission.
- Customers using the "Last Known Location" targeting features are required to upgrade their Braze iOS SDK to at least v3.26.1 in order to accept the new _approximate location_ permission.
- IDFA collection now requires a permission prompt. Failure to update your IDFA collection code will result in a blank value, as if a user declined to provide this permission.
- Apps targeting iOS 14 / Xcode 12 for beta releases can use our [iOS 14 Beta release][1], and our official iOS 14 release after Apple's "Golden Master" release.

### Upgrade Summary

There are two necessary upgrades that are required to continue using Braze messaging features on iOS.

#### Location Targeting

> Customers who are using the _Most Recent Location_ targeting features must upgrade to iOS SDK v3.26.1 which contains updates related to the new _Approximate Location_ feature in iOS 14. Even though your app might not target iOS 14, your end users may upgrade their devices which enable this new location accuracy option. Apps that do not upgrade to iOS SDK v3.26.1+ will not be able to collect location attributes when the _Approximate Location_ is selected by users on iOS 14 devices.

Author: ```suggestion
Customers who are using the _Most Recent Location_ targeting features must upgrade to at least iOS SDK v3.26.1 which contains updates related to the new _Approximate Location_ feature in iOS 14. Even though your app might not target iOS 14, your end users may upgrade their devices which enable this new location accuracy option. Apps that do not upgrade to iOS SDK v3.26.1+ will not be able to collect location attributes when the _Approximate Location_ is selected by users on iOS 14 devices.
```

#### iOS 14 + Xcode 12

> We have released [Beta versions][1] of our upcoming iOS 14 / Xcode 12 compatible SDK. Customers who relase Beta versions of their iOS apps with iOS 14 support are encouraged to use Braze's iOS 14 Beta SDK version which includes Xcode 12 compatibility and support for new iOS 14 APIs. We will continue to release updates and fixes to future beta releases as Apple continues to release newer versions of iOS 14 beta. If you experience any issues or questions related to our iOS 14 compatibility or beta release, please open a new [Github Issue][2].

Author: ```suggestion
We have released [Beta versions][1] of our upcoming iOS 14 / Xcode 12 compatible SDK. Customers who relase Beta versions of their iOS apps with iOS 14 support are encouraged to use Braze's iOS 14 Beta SDK version which includes Xcode 12 compatibility and support for new iOS 14 APIs. 

We will continue to release updates and fixes to future beta releases as Apple continues to release newer versions of iOS 14 beta. If you experience any issues or questions related to our iOS 14 compatibility or beta release, please open a new [Github Issue][2].
```

> Our official iOS 14 compatible SDK will be released shortly after Apple's final iOS 14 beta release, known as the "Golden Master" release.

Author: ```suggestion
Our official iOS 14 compatible SDK will be released shortly after Apple's final iOS 14 beta is released, known as the "Golden Master" release.
```

#### IDFA permissions

> Once users upgrade to iOS 14, apps that collect the IDFA advertising ID must use the new iOS 14 APIs to prompt users for IDFA permission. Failure to update your code will result in a blank IDFA value (`00000000-0000-0000-0000-000000000000`), similar to when a user declines this new prompt.

Author: ```suggestion
Once users upgrade to iOS 14, apps that collect the IDFA advertising ID must use the new iOS 14 APIs to prompt users for IDFA permission. Failure to update your code will result in a blank IDFA value (`00000000-0000-0000-0000-000000000000`), similar to a user declining this new prompt.
```

# iOS 14 SDK Upgrade Guide

This guide describes Braze-related changes introduced in iOS 14 and the required upgrade steps for your Braze iOS SDK integration.

For a complete list of new iOS 14 updates announced this year at WWDC, see Apple's [iOS 14 Preview](https://www.apple.com/ios/ios-14-preview/).

> #### Summary of iOS 14 breaking changes

Reviewer: This is a mix of Braze related changes and iOS related changes, may be good to break apart

Author: 🐃

- Geofences are no longer supported by iOS for users who choose the new  _approximate location_ permission.

> - Customers using the "Last Known Location" targeting features are required to upgrade their Braze iOS SDK to at least v3.26.1 in order to accept the new _approximate location_ permission.

Reviewer: in order to accept the new -> for compatibility with

- IDFA collection now requires a permission prompt. Failure to update your IDFA collection code will result in a blank value, as if a user declined to provide this permission.
- Apps targeting iOS 14 / Xcode 12 for beta releases can use our [iOS 14 Beta release][1], and our official iOS 14 release after Apple's "Golden Master" release.

### Upgrade Summary

There are two necessary upgrades that are required to continue using Braze messaging features on iOS.

#### Location Targeting

Customers who are using the _Most Recent Location_ targeting features must upgrade to at least iOS SDK v3.26.1 which contains updates related to the new _Approximate Location_ feature in iOS 14. 

> Even though your app might not target iOS 14, your end users may upgrade their devices which adds this new location accuracy option. Apps that do not upgrade to iOS SDK v3.26.1+ will not be able to collect location attributes when users provide their _approximate location_  on iOS 14 devices.

Reviewer: will not be able to collect -> will not be able to reliably collect
